src/controllers/video.controller.js

```diff
@@ -1,47 +1,321 @@
-import mongoose, {isValidObjectId} from "mongoose"
-import {Video} from "../models/video.model.js"
-import {User} from "../models/user.model.js"
-import {ApiError} from "../utils/ApiError.js"
-import {ApiResponse} from "../utils/ApiResponse.js"
-import {asyncHandler} from "../utils/asyncHandler.js"
-import {uploadOnCloudinary} from "../utils/cloudinary.js"
+import mongoose, { isValidObjectId } from "mongoose";
+import { deleteOnCloudinary, uploadOnCloudinaary } from "../utils/services/cloudinary.service.js"
+import { Video } from "../models/video.model.js"
+import { asyncHandler } from "../utils/asyncHandler.js"
+import { ApiError } from "../utils/ApiError.js";
+import { ApiResponse } from "../utils/ApiResponse.js";
+import { User } from "../models/user.model.js";
 
+// publish video
+const publishAVideo = asyncHandler(async (req, res) => {
+    const { title , description, isPublished = true } = req.body
 
-const getAllVideos = asyncHandler(async (req, res) => {
-    const { page = 1, limit = 10, query, sortBy, sortType, userId } = req.query
-    //TODO: get all videos based on query, sort, pagination
+    if(!title || title?.trim()===""){
+        throw new ApiError(400, "Title content is required")
+    }
+    if(!description || description?.trim()===""){
+        throw new ApiError(400, "description content is required")
+    }
+    // local path 
+    const videoFileLocalPath = req.files?.videoFile?.[0].path
+    const thumbnailFileLocalPath = req.files?.thumbnail?.[0].path
+
+    if(!videoFileLocalPath){
+        throw new ApiError(400, "video file missing !!")
+    }
+
+    // upload on cloudinary 
+    const videoFile = await uploadOnCloudinaary(videoFileLocalPath)
+    const thumbnail = await uploadOnCloudinaary(thumbnailFileLocalPath)
+
+    if(!videoFile){
+        throw new ApiError(500, "something went wrong while uploading video file on cloudinary")
+    }
+    
+    // strore in the database 
+    const video = Video.create({
```

Comment:
there should be a await

Reply:
Okay, thank you. I will fix that mistake, and I'm sorry for the very late reply. If there are any more mistakes, please let me know now